graylog2-web-interface/src/views/components/searchbar/saved-search/SearchActionsMenu.tsx

```diff
@@ -221,7 +227,9 @@ const SearchActionsMenu = () => {
     callback: () => openFormModal(),
     scope: 'search',
   });
-
+  const streams = filtersToStreamSet(queryFilters.get(currentQuery.id, Immutable.Map())).toJS();
+  const selectedStreamGRN = streams?.map(stream => createGRN('stream', stream));
```

Comment:
one idea (optional) is to extract all this to `useSelectedStreamGRN` hook since this file has already a lot of different logic in it